packages/fuel-gauge/src/transaction.test.ts

```diff
@@ -140,3 +154,167 @@ describe('Transaction', () => {
+    );
+
+    using launched = await setupTestNode();
+    const { privileged, provider } = launched;
+
+    const {
+      chain: { consensusParameters: consensusBeforeUpgrade },
+    } = await provider.fetchChainAndNodeInfo();
+
+    // Update the gas costs to free
+    const { isStatusSuccess, isTypeUpgrade } = await upgradeConsensusParameters(
+      privileged,
+      CONSENSUS_BYTECODE
+    );
+    expect(isStatusSuccess).toBeTruthy();
+    expect(isTypeUpgrade).toBeTruthy();
+
+    // Fetch the upgraded gas costs, they should be different from before
+    const {
+      chain: { consensusParameters: consensusAfterUpgrade },
+    } = await provider.fetchChainAndNodeInfo();
+
+    expect(consensusBeforeUpgrade.gasCosts).not.toEqual(consensusAfterUpgrade.gasCosts);
+  });
+
+  it(
+    'should load bytecode in subsections slowly',
+    async () => {
+      using launched = await setupTestNode();
+      const { privileged, provider } = launched;
+
+      const { subsections, merkleRoot } = subsectionFromBytecode();
+      const requests = subsections.map((subsection) => {
+        const request = UploadTransactionRequest.from({});
+        request.addSubsection({
+          proofSet: subsection.proofSet,
+          subsection: subsection.subsection,
+          root: subsection.root,
+          subsectionsNumber: subsection.subsectionsNumber,
+          subsectionIndex: subsection.subsectionIndex,
+        });
+        return request;
+      });
+
+      // Upload the subsections
+      for (const request of requests) {
+        const cost = await privileged.getTransactionCost(request);
+        request.maxFee = cost.maxFee;
+        await privileged.fund(request, cost);
+        const response = await privileged.sendTransaction(request);
+        const { isTypeUpload, isStatusSuccess } = await response.waitForResult();
+        expect(isTypeUpload).toBeTruthy();
+        expect(isStatusSuccess).toBeTruthy();
+      }
+
+      // Upgrade the chain with the root
+      const request = new UpgradeTransactionRequest();
+      request.addStateTransitionUpgradePurpose(merkleRoot);
+
+      const cost = await privileged.getTransactionCost(request);
+      request.maxFee = cost.maxFee;
+      await privileged.fund(request, cost);
+
+      const response = await privileged.sendTransaction(request);
+      const { isTypeUpgrade, isStatusSuccess, blockId } = await response.waitForResult();
+      expect(isTypeUpgrade).toBeTruthy();
+      expect(isStatusSuccess).toBeTruthy();
+
+      // Check the bytecode version has changed
+      const block = await provider.getBlock(blockId as string);
+      await provider.produceBlocks(1);
+      const nextBlock = await provider.getBlock('latest');
+
+      const versionBeforeUpgrade = block?.header.stateTransitionBytecodeVersion;
+      const versionAfterUpgrade = nextBlock?.header.stateTransitionBytecodeVersion;
+
+      expect(versionBeforeUpgrade).not.toEqual(versionAfterUpgrade);
+    },
+    1000 * 60 * 1
```

Comment:
I had to add a longer timeout because 34 subsections are generated, meaning 34 transactions, and the test was failing on the 10th transaction. This test ended up being a bit large, but that's how the upgrade + upload works.
Is there any problem?

Reply:
Did you try smaller timeouts like 10s or 15s?

One whole minute seems expensive.

I left another suggestion [below](https://github.com/FuelLabs/fuels-ts/pull/3114#discussion_r1748045379).

Reply:
@luisburigo Given the size of each subsection is arbitrary, would it be possible to use a larger subsection size and thus generate less transactions?

Reply:
@maschad Yes, I'm refactoring the test to split it into 192KB subsections. Running the test in isolation took between 7-9 seconds, but running all the tests together caused it to fail.
![image](https://github.com/user-attachments/assets/1c770a37-4ef0-49ba-a5ef-3a843f18c399)

With a 30s timeout, it executed correctly, but the test ran in 40017ms.
![image](https://github.com/user-attachments/assets/fe7c23e2-0ccb-4ec4-861d-bc811a63a23f)